src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -1137,14 +1137,27 @@ protected function callScope($scope, $parameters)
     {
         array_unshift($parameters, $this);
 
-        $query = $this->getQuery();
+        return $this->applyCallback([$this->model, $scope], $parameters);
+    }
+
+    /**
+     * Apply the given callback to a supplied (or the current) builder instance.
+     *
+     * @param  callable $callback
+     * @param  array $parameters
+     * @param  \Illuminate\Database\Query\Builder $query
+     * @return mixed
+     */
+    protected function applyCallback(callable $callback, $parameters = [], $query = null)
```

Comment:
What is the purpose of this method? Could it have a better more descriptive name than `applyCallback` which sounds very vague and general?